### DISCUSSION TITLE
SQL editor query could persist last N commands per database

### DISCUSSION THREAD

**Discussion by bojankoma** [infinum/android_dbinspector#81] [Ideas]
Looking for opinions and feedback on this idea?
Would it be something useful?
If someone needs to shuffle between a few same query commands, this could be a nice thing to have.
In UI, each query would be persisted with a timestamp and flag successful/failed.
Presented to developer in a picker to choose from.
Any thoughts @JonatanPlesko @antunflas @jakoss ?

**Comment by jakoss** [comment]
Totally agree, i'd use that quite a lot

**Comment by JonatanPlesko** [comment]
I think it's useful, but I'm curious about the implementation. I don't think prefilling the existing command area is a good idea. If you want to write a new command, you should be able to do it without having to remove something that was prefilled.
Maybe previously used commands could be a part of the current autocomplete. Or if we can find a separate area where you can access command history in a tap.

**Reply by bojankoma** [reply]
Both implementations entry points are valid and should be an offered option to a developer. 

1. Input area would be enhanced on bottom, suggesting the most similar command/query that you are already typing in. If suggested command is selected - it is executed and/or filled in the input area.

2. Other UI would consist of a button somewhere that opens a Dialog (nicely styled picker) with a list of executed commands. When developer picks one, its still prefilled in input area.

In both cases, editing in input area is fairly easy if you need to modify a part of the already executed and selected command.
I think editing a few characters is a lot more user friendly than typing in the whole command from scratch.

**Comment by bojankoma** [comment]
Enjoy this feature in 5.2.7!
If you experience any issues, please open one.
